src/app/test/test-table/test-table.component.ts

```diff
@@ -0,0 +1,160 @@
+  Input,
+  OnChanges,
+  OnDestroy,
+  OnInit,
+  Output,
+  SimpleChanges,
+} from '@angular/core';
+import { TestListItem } from '../../shared/interfaces/test-list-item';
+import { catchError, Subscription } from 'rxjs';
+import { HttpService } from '../../shared/services/http.service';
+import { ErrorHandling } from '../../shared/classes/error-handling.service';
+import { TabService } from '../../shared/services/tab.service';
+import { HelperService } from '../../shared/services/helper.service';
+import { Report } from '../../shared/interfaces/report';
+import { ReportData } from '../../shared/interfaces/report-data';
+import { View } from '../../shared/interfaces/view';
+import { TestReportsService } from '../test-reports.service';
+import { ToastService } from '../../shared/services/toast.service';
+import { FormsModule } from '@angular/forms';
+
+@Component({
+  selector: 'app-test-table',
+  standalone: true,
+  imports: [FormsModule],
+  templateUrl: './test-table.component.html',
+  styleUrl: './test-table.component.css',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TestTableComponent implements OnChanges, OnInit, OnDestroy {
+  @Input() reports: TestListItem[] = new Array<TestListItem>();
+  @Input() currentFilter: string = '';
+  @Input() showStorageIds?: boolean;
+  @Output() toggleCheckEvent: EventEmitter<TestListItem> = new EventEmitter<TestListItem>();
+  @Output() runEvent: EventEmitter<TestListItem> = new EventEmitter<TestListItem>();
+  protected amountOfSelectedReports: number = 0;
+  private subscriptions: Subscription = new Subscription();
+
+  constructor(
+    private httpService: HttpService,
+    private errorHandler: ErrorHandling,
+    private tabService: TabService,
+    private helperService: HelperService,
+    private testReportsService: TestReportsService,
+    private toastService: ToastService,
+  ) {}
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['currentFilter'] || changes['reports']) {
+      this.amountOfSelectedReports = 0;
+      for (const report of this.reports) {
+        if (report.checked) {
+          this.amountOfSelectedReports++;
+        }
+        if (report.variables) {
+          report.extractedVariables = this.extractVariables(report.variables);
+        }
+      }
+      this.testReportsService.setAmountSelected(this.amountOfSelectedReports);
+      this.getFullPaths();
+    }
+  }
+
+  ngOnInit(): void {
+    this.subscribeToSubscriptions();
+  }
+
+  ngOnDestroy(): void {
+    this.unsubscribeToSubscriptions();
+  }
+
+  subscribeToSubscriptions(): void {
+    const amountSelectedSubscription: Subscription = this.testReportsService.amountSelected$.subscribe({
+      next: (value: number): void => {
+        this.amountOfSelectedReports = value;
+      },
+      error: () => catchError(this.errorHandler.handleError()),
+    });
+    this.subscriptions.add(amountSelectedSubscription);
+  }
+
+  unsubscribeToSubscriptions(): void {
```

Comment:
I know you moved this, but the incorrect English is bothering me 😁
```suggestion
  unsubscribeFromSubscriptions(): void {
```

Reply:
hahaha, don't worry. At other places I used From instead of To. Added the change.